src/main/java/seedu/address/logic/filters/NameFilter.java

```diff
@@ -83,5 +70,33 @@ private static int dist(char[] s1, char[] s2) {
         }
         return prev[s2.length];
     }
+
+    private boolean isSubsequence(char[] s1, char[] s2) {
+        int ptr1 = 0;
+        int ptr2 = 0;
+        while (ptr1 < s1.length && ptr2 < s2.length) {
+            if (s1[ptr1] == s2[ptr2]) {
+                ptr2++;
+            }
+
+            ptr1++;
+        }
+
+        return ptr2 == s2.length;
+    }
+
+    @Override
+    public boolean test(Customer customer) {
+        requireNonNull(customer);
+        String[] customerNameTokens = customer.getName().fullName.split("\\s+");
+        for (String token : customerNameTokens) {
+            for (String possibleName : nameList) {
+                if (levenshteinDistance(token, possibleName)) {
```

Comment:
Also, please rename the function `levenshteinDistance` to `subsequenceDistance`. Its confusing and misleading otherwise.